client/src/components/Alert.tsx

```diff
@@ -0,0 +1,80 @@
+import { storageFactory } from "@justfixnyc/util";
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { forwardRef, useLayoutEffect, useState } from "react";
+import { CloseButton } from "./CloseButton";
+import "styles/_alert.scss";
+
+// TODO: Check with design if we are going to want the icons matching type
+
+export interface AlertProps extends React.ComponentPropsWithoutRef<"div"> {
+  children: React.ReactNode;
+  type?: "error" | "success" | "info";
+  variant?: "primary" | "secondary";
+  closeType?: "none" | "state" | "session" | "local";
+  storageId?: string;
+  className?: string;
+}
+
+export const Alert = forwardRef<HTMLDivElement, AlertProps>(
+  ({ type, variant, closeType, storageId, className, children, ...props }, ref) => {
+    const usingStorage = closeType && ["session", "local"].includes(closeType);
+
+    // TODO: see if there's a way to do this that works with intellisense.
```

Comment:
i think one way to do this would be to have the base `Alert` only do the "none" closetype, then subclass/extend for state, and separately subclass for local/session storage

Reply:
oh cool, that sounds like a better option. I'll look into this more and try that out. thanks!